doc/library/scan.txt

```diff
@@ -383,23 +383,33 @@ Using shared variables - the strict flag
 As we just saw, passing the shared variables to scan may result in a simpler
 computational graph, which speeds up the optimization and the execution. A
 good way to remember to pass every shared variable used during scan is to use
-the ``strict`` flag. When set to true, scan assumes that all the necessary
-shared variables in ``fn`` are passed as a part of ``non_sequences``. This has
-to be ensured by the user. Otherwise, it will result in an error.
+the ``strict`` flag. When set to true, scan checks that all the necessary shared
+variables in ``fn`` are passed as explicit arguments to ``fn``. This has to be
+ensured by the user. Otherwise, it will result in an error.
 
-Using the previous Gibbs sampling example:
+Using the original Gibbs sampling example, with ``strict=True`` added to the
+``scan()`` call:
 
-.. testcode:: scan1
+.. code-block:: python
 
-    # The new scan, using strict=True
-    values, updates = theano.scan(fn=OneStep,
+    # Same OneStep as in original example.
+    def OneStep(vsample) :
+        hmean = T.nnet.sigmoid(theano.dot(vsample, W) + bhid)
+        hsample = trng.binomial(size=hmean.shape, n=1, p=hmean)
+        vmean = T.nnet.sigmoid(theano.dot(hsample, W.T) + bvis)
+        return trng.binomial(size=vsample.shape, n=1, p=vmean,
+                             dtype=theano.config.floatX)
+
+    # The new scan, adding strict=True to the original call.
+    # Produces an error, because W, bvis and bhid are used by OneStep, but not
+    # passed explicitly.
+    values, updates = theano.scan(OneStep,
                                   outputs_info=sample,
-                                  non_sequences=[W, bvis, bhid],
                                   n_steps=10,
                                   strict=True)
 
-If you omit to pass ``W``, ``bvis`` or ``bhid`` as a ``non_sequence``, it will
-result in an error.
+The above will result in an error, indicating that ``OneStep`` relies on
+variables that are not passed as its explicit arguments.
```

Comment:
Can you add the error raised? This way if people google the error, they will find this doc:

MissingInputError: An input of the graph, used to compute InplaceDimShuffle{1,0}(<TensorType(float64, matrix)>), was not provided and not given a value.Use the Theano flag exception_verbosity='high',for more information on this error.

Otherwise, this PR look good. thanks!